NPC's follows the player

JuannFerrari opened the pull request:
This PR is for issue #84

**What**
Adds missing feature related to `follow_player ` var and allows NPC's to follow the player if the value is `true`

**Why**
To let the player set if an NPC will follow him or not

**Demo:**


https://github.com/mapedorr/popochiu/assets/24940330/7752b38a-8338-4ea8-a95f-5283e2c384cd

mapedorr reviewed (changes requested):
Looks fine, but needs some improvements.

JuannFerrari commented on addons/popochiu/engine/objects/room/popochiu_room.gd:
```diff
@@ -131,8 +131,19 @@ func add_character(chr: PopochiuCharacter) -> void:
 	chr.started_walk_to.connect(_update_navigation_path)
 	chr.stopped_walk.connect(_clear_navigation_path.bind(chr))
 	
+	if chr == C.player:
+		chr.started_walk_to.connect(_follow_player)
```
great feedback

JuannFerrari commented:
A new demo addressing the comments, NPC is not flipping its sprite 🤔 

https://github.com/mapedorr/popochiu/assets/24940330/bf080bce-f35f-4984-b027-aa55d88fdb67
